webpack/redux/actions/Settings/tables.js

```diff
@@ -0,0 +1,85 @@
+import { foremanApi, userId } from '../../../services/api';
+
+import {
+  TABLE_COLUMNS_REQUEST,
+  TABLE_COLUMNS_SUCCESS,
+  TABLE_COLUMNS_FAILURE,
+  CREATE_TABLE_COLUMNS,
+  CREATE_TABLE_COLUMNS_SUCCESS,
+  CREATE_TABLE_COLUMNS_FAILURE,
+  UPDATE_TABLE_COLUMNS,
+  UPDATE_TABLE_COLUMNS_SUCCESS,
+  UPDATE_TABLE_COLUMNS_FAILURE,
+} from '../../consts';
+
+const getResponseError = ({ data }) => data && (data.displayMessage || data.error);
+
+export const loadColumns = (table = '') => (dispatch) => {
+  dispatch({ type: TABLE_COLUMNS_REQUEST, params: table });
+
+  return foremanApi
+    .get(`/users/${userId}/table_preferences/${table}`, {})
+    .then(({ data }) => {
+      dispatch({
+        type: TABLE_COLUMNS_SUCCESS,
+        response: data,
+      });
+    })
+    .catch((result) => {
+      const { response } = result;
+      const errorMessage = `Resource table_preference not found by id '${table}'`;
+      if (response.status === 404 && response.data.error.message === errorMessage) {
```

Comment:
Is there a case where a 404 would happen where the error messages don't equal?

Reply:
I removed this altogether from the PR because I changed it to where our loadTables action loads all tables and got rid of the notion of retrieving individual tables.